lib/teamsnap/client.rb

```diff
@@ -0,0 +1,37 @@
+module TeamSnap
+  class Client
+    def initialize
+      yield(configuration) if block_given?
+    end
+
+    def method_missing(method, *args, &block)
+      #TODO: mutex/syncrhonize/just do once
+      TeamSnap.init(connection)
+      TeamSnap.const_get(Inflecto.camelize(method))
```

Comment:
I don't think we should be calling anything on the `TeamSnap::Team` class. We could have instance methods and store `client` as a private `attr_reader` but I don't think we should have any class level methods.